src/__tests__/common.ts

```diff
@@ -162,6 +162,59 @@ export class PaginatedArticleResource extends OtherArticleResource {
   }
 }
 
+export class UnionResource extends Resource {
+  readonly id: string = '';
+  readonly body: string = '';
+  readonly type: string = '';
+
+  pk() {
+    return this.id;
+  }
+
+  static urlRoot = '/union/';
+
+  static detailShape<T extends typeof Resource>(
+    this: T,
+  ): ReadShape<SchemaDetail<AbstractInstanceType<T>>> {
+    const schema = new schemas.Union(
+      {
+        first: FirstUnionResource.getEntitySchema(),
+        second: SecondUnionResource.getEntitySchema(),
+      },
+      'type',
+    );
+    return {
+      ...super.detailShape(),
+      schema,
+    };
+  }
+  static listShape<T extends typeof Resource>(
+    this: T,
+  ): ReadShape<SchemaList<AbstractInstanceType<T>>> {
+    const schema = [
+      new schemas.Union(
+        {
+          first: FirstUnionResource.getEntitySchema(),
+          second: SecondUnionResource.getEntitySchema(),
+        },
+        (input: FirstUnionResource | SecondUnionResource) => input['type'],
```

Comment:
just making sure both forms work here. normally this would be the same as the union for detailShape()